jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/di/DependencyInjectionDiagnosticsCollector.java

```diff
@@ -31,62 +36,165 @@
 import org.eclipse.lsp4jakarta.jdt.core.DiagnosticsCollector;
 import org.eclipse.lsp4jakarta.jdt.core.JDTUtils;
 import org.eclipse.lsp4jakarta.jdt.core.JakartaCorePlugin;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IMethod;
 
 /**
  * 
  * jararta.annotation Diagnostics
  * 
  * <li>Diagnostic 1: @Inject fields cannot be final.</li>
+ * <li>Diagnostic 2: @Inject methods cannot be final.</li>
+ * <li>Diagnostic 3: @Inject methods cannot be abstract.</li>
```

Comment:
Should this description be updated to include that methods cannot be generic or static?